### PR TITLE
add system infos to output when in loglevel debug

### LINKED ISSUE #1530: Can we support reflex to show `system's info` in `--loglevel debug`
**Describe the bug**

The reflex is dependent on many things. 
NodeJS, bun, nvm, OS, PATH, git,  unzip, pythonl, reflex  
But many versions of them may be different.  

If the user can use `reflex --sysinfo` to collect all the correct information quickly.
(By the suggestion, it would be better to show system's information in `--loglevel debug`)

There are 3 good points for this command
- **Bug reproducing**: The User can just paste it when they want to report the bug.  
  It may be helpful for developers to reproduce bugs. 
- **Good for growing up developer community**: It's better to share code on the internet.
  If we have reflex --sysinfo and put the result in the READMD.md, it will help people to build the community and share runnable code easier. 

I saw many examples shared from the showcases channel. 
It looks okay from the video but we cannot run the source code. 
 

 

**To Reproduce**
 - Code/Link to Repo:

**Expected behavior**

**Screenshots**

**Specifics (please complete the following information):**
 - Python Version:
 - Reflex Version:
 - OS: 
 - Browser (Optional):
 
**Additional context**

### PR DESCRIPTION
**Pull request by Lendemor**
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### New Feature Submission:

- [x] Does your submission pass the tests? 
- [x] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Description

Add some output with system info when using `--loglevel debug`

Closes #1530

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -46,6 +46,7 @@ starlette-admin = "^0.9.0"
 python-dotenv = "^0.13.0"
 importlib-metadata = {version = "^6.7.0", python = ">=3.7, <3.8"}
 alembic = "^1.11.1"
+distro = "^1.8.0"
```
Comment (picklelo) [inline pyproject.toml:49]: Not sure if this is worth bringing in a new dependency to the project

File: reflex/utils/exec.py [hunk 2]
```diff
@@ -155,3 +157,50 @@ def run_backend_prod(
...
+    for dep in dependencies:
+        console.debug(f"{dep}")
+
+    if system == "Windows":
+        console.debug(
+            f"Using package manager at : {prerequisites.get_windows_package_manager()}"
+        )
+    else:
+        console.debug(
+            f"Using package installer at : {prerequisites.get_install_package_manager()}"
```
Comment (picklelo) [inline reflex/utils/exec.py:200]: This method handles the Windows case also

Comment (Lendemor) [reply reflex/utils/exec.py:200]: I see.
But we still need different output anyway because Windows use npm for both.

File: pyproject.toml (hunk 1 above)
Comment (masenf) [reply pyproject.toml:49]: can we add something like 
```suggestion
distro = {version = "^1.8.0", platform = "linux"}
```

i'm not convinced that `distro` is a heavy dependency, but it's not going to be used at all on the other platforms...

File: reflex/utils/exec.py (hunk 2 above)
Comment (masenf) [reply reflex/utils/exec.py]: refactored this now that `get_windows_package_manager` is no longer used

**Comment by masenf** [comment]
# mac

```console
───────────────────────────────────── System Info ──────────────────────────────────────
Debug: Running command: ['/Users/masenf/.reflex/.nvm/versions/node/v18.17.0/bin/node', 
'-v']
Debug: Running command: ['/Users/masenf/.reflex/.bun/bin/bun', '-v']
Debug: [Reflex 0.2.3 with Python 3.11.4 (PATH: 
/Users/masenf/Library/Caches/pypoetry/virtualenvs/reflex-1Q8YAZ6V-py3.11/bin/python)]
Debug: [Node 18.17.0 (Expected: 18.17.0) 
(PATH:/Users/masenf/.reflex/.nvm/versions/node/v18.17.0/bin/node)]
Debug: [NVM 0.39.1 (Expected: 0.39.1) (PATH: /Users/masenf/.reflex/.nvm/nvm.sh)]
Debug: [Bun 0.7.0 (Expected: 0.7.0) (PATH: /Users/masenf/.reflex/.bun/bin/bun)]
Debug: [OS Darwin Darwin Kernel Version 22.6.0: Wed Jul  5 22:21:53 PDT 2023; 
root:xnu-8796.141.3~6/RELEASE_ARM64_T6020]
Debug: Using package installer at: /Users/masenf/.reflex/.bun/bin/bun
Debug: Using package executer at: 
/Users/masenf/.reflex/.nvm/versions/node/v18.17.0/bin/npm
Debug: Unzip path: /usr/bin/unzip
```

# windows

```console
───────────────────────────────────────────────────── System Info ──────────────────────────────────────────────────────
Debug: Running command:
['C:\\Users\\masenf\\AppData\\Local\\reflex\\fnm\\node-versions\\v18.17.0\\installation\\node.exe', '-v']
Debug: [Reflex 0.2.4 with Python 3.11.4 (PATH:
C:\Users\masenf\AppData\Local\pypoetry\Cache\virtualenvs\reflex-EM2WRxPK-py3.11\Scripts\python.exe)]
Debug: [Node 18.17.0 (Expected: 18.17.0)
(PATH:C:\Users\masenf\AppData\Local\reflex\fnm\node-versions\v18.17.0\installation\node.exe)]
Debug: [FNM 1.35.1 (Expected: 1.35.1) (PATH: C:\Users\masenf\AppData\Local\reflex\fnm\fnm.exe)]
Debug: [OS Windows 10.0.22621]
Debug: Using package installer at: C:\Users\masenf\AppData\Local\reflex\fnm\node-versions\v18.17.0\installation\npm
Debug: Using package executer at: C:\Users\masenf\AppData\Local\reflex\fnm\node-versions\v18.17.0\installation\npm
```

# linux

## pre-init

```console
───────────────────────────────── System Info ──────────────────────────────────
Debug: Running command: ['/root/.reflex/.nvm/versions/node/v18.17.0/bin/node', 
'-v']
Debug: Running command: ['/root/.reflex/.bun/bin/bun', '-v']
Debug: [Reflex 0.2.4 with Python 3.11.4 (PATH: 
/root/.cache/pypoetry/virtualenvs/reflex-XBQjXVTQ-py3.11/bin/python)]
Debug: [Node None (Expected: 18.17.0) 
(PATH:/root/.reflex/.nvm/versions/node/v18.17.0/bin/node)]
Debug: [NVM 0.39.1 (Expected: 0.39.1) (PATH: /root/.reflex/.nvm/nvm.sh)]
Debug: [Bun None (Expected: 0.7.0) (PATH: /root/.reflex/.bun/bin/bun)]
Debug: [OS Linux Debian GNU/Linux 12 (bookworm)]
Debug: Using package installer at: /root/.reflex/.bun/bin/bun
Debug: Using package executer at: 
/root/.reflex/.nvm/versions/node/v18.17.0/bin/npm
Debug: Unzip path: /usr/bin/unzip
```

## post-init
```console
───────────────────────────────── System Info ──────────────────────────────────
Debug: Running command: ['/root/.reflex/.nvm/versions/node/v18.17.0/bin/node', 
'-v']
Debug: Running command: ['/root/.reflex/.bun/bin/bun', '-v']
Debug: [Reflex 0.2.4 with Python 3.11.4 (PATH: 
/root/.cache/pypoetry/virtualenvs/reflex-XBQjXVTQ-py3.11/bin/python)]
Debug: [Node 18.17.0 (Expected: 18.17.0) 
(PATH:/root/.reflex/.nvm/versions/node/v18.17.0/bin/node)]
Debug: [NVM 0.39.1 (Expected: 0.39.1) (PATH: /root/.reflex/.nvm/nvm.sh)]
Debug: [Bun 0.7.0 (Expected: 0.7.0) (PATH: /root/.reflex/.bun/bin/bun)]
Debug: [OS Linux Debian GNU/Linux 12 (bookworm)]
Debug: Using package installer at: /root/.reflex/.bun/bin/bun
Debug: Using package executer at: 
/root/.reflex/.nvm/versions/node/v18.17.0/bin/npm
Debug: Unzip path: /usr/bin/unzip
```

File: reflex/utils/exec.py [hunk 3]
```diff
@@ -154,3 +156,50 @@ def run_backend_prod(
...
+    for dep in dependencies:
+        console.debug(f"{dep}")
+
+    console.debug(
+        f"Using package installer at: {prerequisites.get_install_package_manager()}"
+    )
+    console.debug(f"Using package executer at: {prerequisites.get_package_manager()}")
+    if system != "Windows":
+        console.debug(f"Unzip path: {path_ops.which('unzip')}")
+    # exit()
```
Comment (picklelo) [inline reflex/utils/exec.py:205]: remove this?

File: reflex/utils/prerequisites.py [hunk 4]
```diff
@@ -31,20 +31,30 @@ def check_node_version() -> bool:
...
     """
+    current_version = get_node_version()
+    if current_version:
+        # Compare the version numbers
+        return (
+            current_version >= version.parse(constants.NODE_VERSION_MIN)
+            if constants.IS_WINDOWS
+            else current_version == version.parse(constants.NODE_VERSION)
+        )
+    else:
```
Comment (picklelo) [inline reflex/utils/prerequisites.py:42]: Don't need the `else` here

**Review by picklelo — APPROVED** [review]
looks good, just nits we can clean up in a follow up

**Comment by milochen0418** [comment]
Nice~

**Pull request merged**